### PR TITLE
develop: docker migration to php 8.1

### PR DESCRIPTION
**Pull request by fliot**

### CODE REVIEW

**Comment by alextselegidis** [comment]
Hello! 

Thanks for this PR, I'll review and get back to you. 

<small>
<table>
<tr>
<td></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by alextselegidis** [comment]
Thanks for the great proposals! I've posted a couple of questions for you to check. 

<small>
<table>
<tr>
<td></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Comment by fliot** [comment]
Hi @alextselegidis,
I don't see any comment or question, anywhere.
Not able to review or answer them.
Let me know your thoughs.
Regards.

**Comment by alextselegidis** [comment]
Hi @fliot 

Aren't the following review comments visible to you? 

image

**Comment by fahim44** [comment]
@alextselegidis your comments are in the pending state. You should find a submit button below.

**Comment by fliot** [comment]
Hi @alextselegidis ,
I confirm, I do not see yet your comments,
You probably have to submit them, as described by @fahim44 
Thanks in advance.

**Review by alextselegidis — CHANGES_REQUESTED** [review]

**Comment by alextselegidis** [comment]
Thanks, that was it indeed 👍 

<small>
<table>
<tr>
<td></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

File: docker/php-fpm/Dockerfile [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
+
+RUN apt-get install -y --no-install-recommends -q ssmtp mailutils \
     && echo "hostname=localhost.localdomain" > /etc/ssmtp/ssmtp.conf \
     && echo "root=root@presta.com" >> /etc/ssmtp/ssmtp.conf \
     && echo "mailhub=mailhog:1025" >> /etc/ssmtp/ssmtp.conf \
-    && echo "sendmail_path=/usr/sbin/ssmtp -t" >> /usr/local/etc/php/conf.d/php-sendmail.ini \
-    && apt-get -y autoremove \
+    && echo "sendmail_path=/usr/sbin/ssmtp -t" >> /usr/local/etc/php/conf.d/php-sendmail.ini
+
+RUN apt-get -y autoremove \
```
[the remark this answers is not in the document]
Comment (fliot) [reply docker/php-fpm/Dockerfile:24]: Lighter result, just removing 100 Mb to the final image.

File: application/config/routes.php [hunk 2]
```diff
@@ -77,26 +77,17 @@
 | -------------------------------------------------------------------------
 | CORS HEADERS
 | -------------------------------------------------------------------------
-| Set the appropriate headers so that CORS requirements are met and any 
-| incoming preflight options request succeeds. 
+| The accepted 3rd party websites, for which CORS must be configured,
```
[the remark this answers is not in the document]
Comment (fliot) [reply application/config/routes.php:80]: Several points,
1) your default proposal (before my patch) allows "*" for cors, so for most of usage, it's an immediate working configuration, but it's clearly unsecure
2) your initial code wasn't perfectly working with HEAD (which is used by many javascript rest implementation)
3) your initial code wasn't reporting allow-credentials and allow-headers directives (see docker/nginx/nginx.conf) which is also for many javascript implementation a real issue.

So, about a default configuration,
I fully agree with you, most users won't be able to handle this by-themselves.
The cleanest way to handle this, is to keep track of this in the database, and authorize backend users to configure cors authorized domains directly.

Only "origin" must be configurable, other headers must be hardcoded :
Access-Control-Allow-Origin $http_origin # with default value "*" to not break compatibility
Access-Control-Allow-Credentials "true"
Access-Control-Allow-Methods "GET,POST,PUT,PATCH,DELETE,HEAD,OPTIONS"
Access-Control-Allow-Headers "Authorization, Access-Control-Allow-Headers, Origin, Accept, X-Requested-With, Content-Type, Access-Control-Request-Method, Access-Control-Request-Headers"

**Comment by alextselegidis** [comment]
Hello! 

Quick update on this PR, there are some parts that I would definitely like to add to the current set up. 

The manual merge will take place soon. 

Thanks for the great work!

<small>
<table>
<tr>
<td></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

File: application/config/routes.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (alextselegidis) [reply application/config/routes.php:80]: Thanks for all the clarifications, I really appreciate them. 

For now and in order to keep things as simple as possible for the most people, I will keep the inline CORS configuration and document in the updated developers docs (will be published soon). 

I will not make this a setting yet, most likely will do so in the future.

**Comment by alextselegidis** [comment]
Hello! 

I have now migrated all the parts that I'd like to have in the developer docker-compose setup. 

Thanks for this PR! 

<small>
<table>
<tr>
<td></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>

**Pull request closed without merging**